python/ingestion/dataset_utils.py

```diff
@@ -492,3 +492,27 @@ def zero_out_pct_rel_inequity(df: pd.DataFrame,
         ), pct_inequity_col] = np.nan
 
     return df
+
+
+def remove_non_current_rows(df: pd.DataFrame, keep_time_period_col: bool = False):
+    """ Takes a dataframe with a `time_period` col that contains datatime strings
```

Comment:
Can we make the function name a bit more descriptive? `filter_most_recent_time_period_rows` or something like that.